src/User.php

```diff
@@ -558,7 +558,9 @@ public function getFromDBbySyncField($value)
      */
     public function getFromDBbyDn($user_dn)
     {
-        return $this->getFromDBByCrit(['user_dn' => $user_dn]);
+        return $this->getFromDBByCrit([
+            'user_dn_hash' => hash('sha256', $user_dn)
```

Comment:
Why not using a simple `sha1($user_dn)` ? We just want to have a smaller indexable value here, and we store the clear value in the `user_dn` field, so there is no need to use a secured hashing algo here. If you use `sha1()` or `md5()`, you will be able to remove the requirement of the `hash` extension that make `composer-require-checker` unhappy.